frontend/ui/widget/touchmenu.lua

```diff
@@ -928,6 +928,7 @@ function TouchMenu:onMenuSelect(item, tap_on_checkmark)
             end
         else
             table.insert(self.item_table_stack, self.item_table)
+            item.menu_item_id = item.menu_item_id or os.time() -- unique id
```

Comment:
If I understand that right, menu items get to have a menu_item_id only when they are entered, so when going up and iterating the parent items, we'll only meet ones with that menu_item_id that we have entered, and so it will work.
No risk with os.time() having a granularity of the second, and someone entering/leaving multiple submenus in the same second? No better random and unique id ? ie. the id/address of the table:
```
> a={1,2}
> print(tostring(a))
table: 0x7f71bcd83788
```
or some more granular time from UIManager. Pinging @NiLuJe .

Reply:
The gc ref of the widget actually sounds like a better idea, yeah!

Reply:
Unless it needs to change on each entry?

Reply:
In which case; we have [random.uuid](https://github.com/koreader/koreader/blob/ab47bd10a4b65a7ad5161b80fbcbbbf03689c2aa/frontend/random.lua#L14)